protocol/dubbo/dubbo_invoker.go

```diff
@@ -62,6 +63,8 @@ func NewDubboInvoker(url common.URL, client *Client) *DubboInvoker {
 
 // Invoke ...
 func (di *DubboInvoker) Invoke(ctx context.Context, invocation protocol.Invocation) protocol.Result {
+	di.RLock()
```

Comment:
so large lock scope.

Reply:
Before `di.RLock()`, I suggest  to judge where this is be destroyed or not. If already being destroyed, func should return error soon.